pkg/trace/stats/otel_util.go

```diff
@@ -0,0 +1,157 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package stats
+
+import (
+	"slices"
+	"sync"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/traceutil"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.17.0"
+)
+
+// chunkKey is used to group TraceChunks
+type chunkKey struct {
+	traceIDUInt64 uint64
+	env           string
+	version       string
+	hostname      string
+	cid           string
+}
+
+var containerTagsByIDPool = &sync.Pool{
+	New: func() interface{} {
+		return make(map[string][]string)
+	},
+}
+
+func getContainerTagsByIDFromPool() map[string][]string {
+	return containerTagsByIDPool.Get().(map[string][]string)
+}
+
+func putContainerTagsByIDToPool(containerTagsByID map[string][]string) {
+	clear(containerTagsByID)
+	containerTagsByIDPool.Put(containerTagsByID)
+}
```

Comment:
Shouldn't the pool and the associated funcs be in `concentrator.go`? I don't think they're too specific to "OTel util".